src/Interactive/Host/Interactive/Core/InteractiveHost.cs

```diff
@@ -413,25 +369,38 @@ private async Task<InitializedRemoteService> TryGetOrCreateRemoteServiceAsync()
             return default;
         }
 
-        private async Task<TResult> Async<TResult>(Action<Service, RemoteAsyncOperation<TResult>> action)
+        private async Task<TResult> Async<TResult>(string targetName, params object[] arguments)
         {
+            var initializedService = await TryGetOrCreateRemoteServiceAsync().ConfigureAwait(false);
+            TaskCompletionSource<TResult> _completion = new TaskCompletionSource<TResult>();
+            EventHandler _processExitedHandler = new EventHandler((_, __) =>
+            {
+                initializedService.ServiceOpt.Process.Exited -= _processExitedHandler;
+                _completion.TrySetResult(default(TResult));
+            });
+
             try
             {
-                var initializedService = await TryGetOrCreateRemoteServiceAsync().ConfigureAwait(false);
-                if (initializedService.ServiceOpt == null)
+                initializedService.ServiceOpt.Process.Exited += _processExitedHandler;
+                if (!initializedService.ServiceOpt.Process.IsAlive())
                 {
-                    return default;
+                    initializedService.ServiceOpt.Process.Exited -= _processExitedHandler;
+                    _completion.TrySetResult(default(TResult));
                 }
```

Comment:
We can remote this also.

Reply:
When you say remote this as well, can we just remove it from the method or refactor it for the rpc?

Reply:
typo - I meant remove :)